Assets/Scripts/FPSCapper.cs

```diff
@@ -0,0 +1,34 @@
+﻿using UnityEngine;
+using System.Threading;
+
+namespace Assets.Scripts
+{
+    public class FpsCapper : MonoBehaviour
+    {
+        float oldTime = 0.0F;
+        float theDeltaTime = 0.0F;
+
+        // Use this for initialization
+        void Start()
+        {
+            int frameRate = Screen.currentResolution.refreshRate;
+            theDeltaTime = (1.0F / frameRate);
+            oldTime = Time.realtimeSinceStartup;
+        }
+
+
+        // Update is called once per frame
+        void LateUpdate()
+        {
+            float curTime = Time.realtimeSinceStartup;
+            float timeTaken = (curTime - oldTime);
+            if (timeTaken < theDeltaTime)
+            {
+                Thread.Sleep((int) (1000 * (theDeltaTime - timeTaken)));
```

Comment:
Thread.Sleep should never be used in Synchronous code as it blocks the entire thread. If you want to wait a specific amount of time, you should rather use Coroutines (https://docs.unity3d.com/Manual/Coroutines.html) if you really have to wait.

Although, what does this code actually do? It sets oldTime and theDeltaTime variables, but those are never used in the application